sql-plugin/src/main/scala/com/nvidia/spark/rapids/RegexParser.scala

```diff
@@ -689,6 +681,8 @@ sealed class RegexRewriteFlags(val emptyRepetition: Boolean)
 class CudfRegexTranspiler(mode: RegexMode) {
   private val regexMetaChars = ".$^[]\\|?*+(){}"
   private val regexPunct = "!\"#$%&'()*+,-./:;<=>?@[\\]^_`{|}~"
+  private val escapeChars = Map('n' -> '\n', 'r' -> '\r', 't' -> '\t', 'f' -> '\f', 'a' -> '\u0007',
+      'b' -> '\b', 'e' -> '\u001b')
```

Comment:
Could we define this map once and reference it from `RegexParser` and `CudfRegexTranspiler`?